tools/validators-monitor/epoch/epoch.go

```diff
@@ -0,0 +1,84 @@
+// internal/epoch/epoch.go
+package epoch
+
+import (
+	"fmt"
+	"time"
+)
+
+// Constants for Ethereum 2.0 timing
+const (
+	// Genesis time for mainnet in Unix timestamp
+	MainnetGenesisTime = 1606824023
+)
+
+// Calculator handles epoch and slot calculations
+type Calculator struct {
```

Comment:
Maybe we can move this into 'x' or 'util' pkg and reuse it in mev-commit validator API code as well.

Reply:
Maybe it’s better to do it in a separate PR? This one is already quite big.